controller/upload_file.rst

```diff
@@ -118,25 +123,26 @@ Finally, you need to update the code of the controller that handles the form::
             $form->handleRequest($request);
 
             if ($form->isSubmitted() && $form->isValid()) {
-                // $file stores the uploaded PDF file
-                /** @var Symfony\Component\HttpFoundation\File\UploadedFile $file */
-                $file = $product->getBrochure();
-
-                $fileName = $this->generateUniqueFileName().'.'.$file->guessExtension();
+                /** @var UploadedFile $brochureFile */
+                $brochureFile = $form['brochure']->getData();
+                $originalFilename = pathinfo($brochureFile->getClientOriginalName(), PATHINFO_FILENAME);
+                // this is needed to safely include the file name as part of the URL
+                $originalFilename = transliterator_transliterate('Any-Latin; Latin-ASCII; Lower()', $originalFilename);
```

Comment:
I’ve never seen this transliterator_transliterate 😳

Reply:
It's been part of PHP for a long time. I've made some improvements in the code:

```php
$input = 'Iñtërnâtiônàlizætiøn !"·$%&/()=? 😊';
$output = transliterator_transliterate('Any-Latin; Latin-ASCII; [^A-Za-z0-9_] remove; Lower()', $input);
// $output = 'internationalizaetion';
```